docs/concepts/web3/nfts.md

```diff
@@ -0,0 +1,643 @@
+2. Timestamp verification - prevents replay attacks. Server can verify that the request’s timestamp is not too old (e.g. has been created no more than 10 seconds ago).
+3. Public key ownership verification - by calling [view_access_key method](https://docs.near.org/docs/api/rpc/access-keys#view-access-key), we can make sure that the provided public key is really associated with the provided account.
+
+Such authentication approach is the simplest one, but has a few major drawbacks:
+
+
+
+* Performing a REST API call to RPC Node is quite expensive to do each time from the performance perspective.
+* We can’t sign requests from the mobile client, since it usually should be disconnected from the blockchain due to store policies, and hence doesn’t have a key pair.
+* A NEAR account is required in order to start using the application, which complicates the onboarding process.
+
+To solve the first problem, we can simply issue a JWT token or authenticate connection in some other way after a successful NEAR account authentication, so the it will serve as “login” of sorts:
+
+
+
+![image](/docs/assets/web3/nfts-8.png)
+
+
+
+
+While this may be enough for some applications, it doesn’t address the last 2 problems. In order to solve all of them, we can use a hybrid authentication approach with 2 accounts:
+
+
+
+1. “Classic” Web 2 account - all clients can use this account to call a server. For example, this can be a simple username/password or OAuth 2 login with a JWT token.
+2. NEAR account - can be used from non-mobile clients only. Instead of performing NEAR account auth each time we need to use it, we can do it a single time in order to “connect” this account to our primary Web 2 account and store Classic-NEAR account connection in our server database. In this way we solve all problems - server doesn’t need to authenticate NEAR account each time it wants to perform an authorization, instead it can read an associated NEAR account from its own database.
+
+With such hybrid approach, different authentication methods are used for blockchain and server:
+
+
+
+![image](/docs/assets/web3/nfts-9.png)
+
+
+
+NEAR account connection sequence can be implemented in a very similar way to the already described NEAR authentication method, where at the end we store an authenticated account in our database:
+
+
+
+![image](/docs/assets/web3/nfts-10.png)
+
+
+
+
+There’s one more improvement we can make to this flow. Since a  Web Client uses both accounts, a user is forced to login using both Web 2 login method (e.g. login/password) and NEAR Wallet. This is not ideal from the UX perspective, so we can simplify it by introducing a “Login with Wallet” method to our server, which would work when a user already has a wallet connected. We can do this in a similar way to the account connection flow:
+
+
+
+![image](/docs/assets/web3/nfts-11.png)
```

Comment:
Current image has a small error that I've fixed in the original document. Please update the diagram.

Reply:
image replaced